### PR TITLE
chore: add user deletion stats for regulation worker at destType level

### PR DESCRIPTION
**Pull request by anantjain45823**
## What are the changes introduced in this PR?

Linear: Resolves INT-1149
Adding metrics for regulation worker user deletion requests on `destType` level

## Please explain the objectives of your changes below
Currently we do not have any metrics for user deletion / regulation worker requests to find count of requests destination wise.
So we need to add metric for the same in nativeIntegration.ts
(P.S. We are not adding it in regulation.ts as it contains array of all these requests which can contain requests of different destType)

Also for failure count for userDeletion module there is TODO item as well. So picking it up in the same

**Metric name**: `regulation_worker_requests_dest_latency`

**Labels**: `destType, feature, implementation`

**Metric name**: `regulation_worker_user_deletion_failure`

**Labels**: `'destType', 'module', 'implementation', 'feature'`

### Type of change

chore

- Existing capabilities/behavior -> Had metrics on latency in total for user deletion requests independent of `destType`

- New capabilities/behavior-> New metric for count, latency and failure count is added on `destType` tag

<hr>

### Developer checklist

- [x] **No breaking changes are being introduced.**

- [x] Are all related docs linked with the PR?

- [x] Are all changes manually tested?

- [ ] Does this change require any documentation changes?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [x] Is the type of change in the PR title appropriate as per the changes?

- [x] Verified that there are no credentials or confidential data exposed with the changes.

## Summary by CodeRabbit

- **New Features**
  - Introduced new metrics to monitor user deletion failures and request latencies.

- **Improvements**
  - Enhanced tracking capabilities with additional labels for detailed metric information.

- **Bug Fixes**
  - Implemented tracking for user deletion failures due to regulation worker errors.

### CODE REVIEW

File: src/services/destination/nativeIntegration.ts [hunk 1]
```diff
@@ -203,6 +204,7 @@ export class NativeIntegrationDestinationService implements DestinationService {
   ): Promise<UserDeletionResponse[]> {
     const response = await Promise.all(
       requests.map(async (request) => {
+        const startTime = new Date();
```
[the remark this answers is not in the document]
Comment (yashasvibajpai) [reply src/services/destination/nativeIntegration.ts:207]: @anantjain45823 worth looking.

**Review by yashasvibajpai — APPROVED** [review]

File: src/services/destination/nativeIntegration.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (anantjain45823) [reply src/services/destination/nativeIntegration.ts:207]: @coderabbitai it is already inside the map callback function where there would be only one destType

[the remark this answers is not in the document]
Comment (anantjain45823) [reply src/services/destination/nativeIntegration.ts:207]: Yeah so can we resolve this

**Review by sanpj2292 — APPROVED** [review]

File: src/services/destination/nativeIntegration.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (anantjain45823) [reply src/services/destination/nativeIntegration.ts:207]: Resolving this as no action is needed here

**Pull request merged**